pkg/ingester/wal.go

```diff
@@ -464,6 +458,23 @@ func (w *walWrapper) deleteCheckpoints(maxIndex int) (err error) {
 	return errs.Err()
 }
 
+var checkpointRe = regexp.MustCompile("^" + checkpointPrefix + "(\\d+)(\\.tmp)?$")
```

Comment:
While this works, `checkpointPrefix` is `checkpoint.`, while it should be `checkpoint\\.` for the regex. Else it also matches something like `checkpoints12345`.